R/Features-filter.R

```diff
@@ -189,3 +207,13 @@ filterFeaturesWithFormula <- function(object, filter, ...) {
                   })
     object[sel, , ]
 }
+
+
+## Internal function called by `filterFeaturesWithAnnotationFilter` when 
+## `condition` is `"contains"`
+contains <- function(x, value){
```

Comment:
If this is an internal function, could you rename if `.contains`.

Reply:
If I'm not mistaken, the user then has to specify `condition == ".contains"` which is not accepted by `AnnotationFilter`. 
* Maybe I should export it with some documentation like in `?base::startsWith`. I just think it's weird that `Features` would export a function that has nothing to do with MS.
* Maybe I should create an issue in `AnnotationFilter` or ask @jorainer ?

Reply:
@jorainer - could you help us with the `contains` condition. How does the `contains` and `%like%` work, where are they implemented. Is it normal that we need to implement our own `contains` function?

Reply:
The `AnnotationFilter` package does not do any filtering itself, it just defines the filter classes and allowed comparisons (basically it defines the API for the filtering).

The package using the filter has to implement all the logic to perform the filtering. Example `ensembldb`: I'm supporting `AnnotationFilter`s in that package and you can either pass them to the `genes`, `transcripts` ... function with the parameter `filter`, or you can use the `filter` method on the database itself. Internally, I'm converting the filter into the SQL query to extract data from the database. I have a method `ensDbQuery` that does this conversion (https://github.com/jorainer/ensembldb/blob/master/R/Methods-Filter.R) and uses internal functions in https://github.com/jorainer/ensembldb/blob/master/R/functions-Filter.R to actually *translate* the filter expressions to SQL.

Summarizig, I guess you will have to implement your own filter functions, and along with that you have to implement the way how you will want to deal with `"contains"` - the `%like%` is actually an SQL command - you will most likely want to use a `grepl` or similar (if you're doing string comparisons in plain R).